crypto/evp/evp_lib.c

```diff
@@ -1226,4 +1226,31 @@ EVP_PKEY *EVP_PKEY_Q_keygen(OSSL_LIB_CTX *libctx, const char *propq,
     return ret;
 }
 
+EVP_PKEY *EVP_PKEY_generate_public_key(OSSL_LIB_CTX *libctx, const char *propq,
```

Comment:
It would also be useful to have a function to test whether a key has a missing public or private key component. Much like `EVP_PKEY_missing_parameters` already does.

Reply:
I can add that, although imo we could just make the evp_keymgmt_util_has() public. Not sure if it is a problem that it will support only provided keys.

Reply:
IMO this is for a separate PR